scalafmt-tests/src/test/scala/org/scalafmt/util/DiffAssertions.scala

```diff
@@ -14,10 +16,11 @@ trait DiffAssertions extends FunSuiteLike {
   case class DiffFailure(title: String,
                          expected: String,
                          obtained: String,
-                         diff: String)
+                         diff: String)(implicit pos: Position)
       extends TestFailedException(
-        title + "\n" + error2message(obtained, expected),
-        1)
+        _ => Some(title + "\n" + error2message(obtained, expected)),
+        None: Option[Throwable],
+        pos)
```

Comment:
This looks like a big change but it produces the same output (except that now it will point to the place in CliTests where it failed, not to this file). It just looks this bad because `TestFailedException` has an odd interface (it has many overloaded `this` but it doesn't (surprisingly) have the basic, and probably common, "a message and a Position").